plugins/mtd/fu-mtd-device.c

```diff
@@ -293,19 +372,35 @@ fu_mtd_device_probe(FuDevice *device, GError **error)
 	if (attr_name != NULL)
 		fu_device_set_name(FU_DEVICE(self), attr_name);
 
-	/* set vendor ID as the BIOS vendor */
-	vendor = fu_context_get_hwid_value(ctx, FU_HWIDS_KEY_MANUFACTURER);
-	fu_device_build_vendor_id(device, "DMI", vendor);
+	device_subsys_fn = g_build_filename(fu_udev_device_get_sysfs_path(FU_UDEV_DEVICE(device)),
+					    "device",
+					    "subsystem",
+					    NULL);
+	device_subsys_target = g_file_read_link(device_subsys_fn, NULL);
```

Comment:
can you use `fu_udev_device_get_symlink_target` here? We need to use the wrappers so we can emulate devices.